invalid consensus message: missing blockId

Encountered this error using `tmkms` live on gaia-9002 (first time I've been able to use it on a testnet):

```
22:48:06 [DEBUG] tmkms::session: started handling request ...
22:48:06 [ERROR] [gaia-9002@tcp://10.10.8.10:5007] invalid consensus message: missing blockId
22:48:11 [INFO] KMS node ID: 1B234585630E2CF85D7972D91FE8D849F911022C
22:48:11 [DEBUG] tmkms::session: gaia-9002: Connecting to 10.10.8.10:5007...
```

FWIW, I've been planning on moving `tmkms` to do structural logging (via [slog](https://crates.io/crates/slog)) which I think would be helpful in debugging these sorts of issues.

@ebuchman yeah, the KMS is closing the connection after `tmkms` errors out with:

```
[ERROR] [gaia-9002@tcp://10.10.8.10:5007] invalid consensus message: missing blockId
```

--- a/tendermint-rs/src/amino_types/block_id.rs
+++ b/tendermint-rs/src/amino_types/block_id.rs
@@ -25,7 +25,6 @@ impl ConsensusMessage for BlockId {
         if !self.hash.is_empty() && self.hash.len() != SHA256_HASH_SIZE {
             return Err(InvalidHashSize.into());
         }
-        // TODO: is an empty PartsSetHeader really OK here?
         self.parts_header
             .as_ref()
             .map_or(Ok(()), |psh| psh.validate_basic())

--- a/tendermint-rs/src/amino_types/validate.rs
+++ b/tendermint-rs/src/amino_types/validate.rs
@@ -25,8 +25,6 @@ pub enum ValidationErrorKind {
     InvalidHashSize,
     #[fail(display = "negative total")]
     NegativeTotal,
-    #[fail(display = "missing blockId")]
-    MissingBlockId,
 }
 
 impl ToString for ValidationError {

--- a/tendermint-rs/src/amino_types/vote.rs
+++ b/tendermint-rs/src/amino_types/vote.rs
@@ -172,10 +172,10 @@ impl ConsensusMessage for Vote {
         if self.validator_address.len() != VALIDATOR_ADDR_SIZE {
             return Err(InvalidValidatorAddressSize.into());
         }
-        match self.block_id {
-            Some(ref bid) => bid.validate_basic(),
-            None => Err(MissingBlockId.into()),
+        if let Some(ref bid) = self.block_id {
+            return bid.validate_basic();
         }
+        Ok(())
         // signature will be missing as the KMS provides it
     }
 }

--- a/tendermint-rs/src/amino_types/vote.rs
+++ b/tendermint-rs/src/amino_types/vote.rs
@@ -172,10 +172,11 @@ impl ConsensusMessage for Vote {
         if self.validator_address.len() != VALIDATOR_ADDR_SIZE {
             return Err(InvalidValidatorAddressSize.into());
         }
-        if let Some(ref bid) = self.block_id {
-            return bid.validate_basic();
-        }
-        Ok(())
+
+        self.block_id
+            .as_ref()
+            .map_or(Ok(()), |bid| bid.validate_basic())
+
         // signature will be missing as the KMS provides it
     }
 }

